datafusion/core/src/physical_plan/sorts/sort.rs

```diff
@@ -285,14 +286,13 @@ impl ExternalSorter {
             })
             .collect::<Result<_>>()?;
 
-        // TODO: Pushdown fetch to streaming merge (#6000)
-
         streaming_merge(
             streams,
             self.schema.clone(),
             &self.expr,
             metrics,
             self.batch_size,
+            self.fetch,
```

Comment:
This was moved in a PR to use `streaming_merge` in `SortExec` causing `SortExec` to spill even with a (low) `fetch`.